src/scrcmd_move_tutor.c

```diff
@@ -0,0 +1,335 @@
+    numLearnableMoves = GetLearnableTutorMoves(mon, moveTutorNpc, learnableMoves);
+    s32 numMovesToSkip;
+    if (numLearnableMoves <= 7) {
+        numMovesToSkip = 0;
+        showNextButton = FALSE;
+    } else if (numLearnableMoves > pageNum * 6) {
+        numMovesToSkip = pageNum * MAX_TUTOR_MOVES_PER_PAGE;
+        numLearnableMoves -= numMovesToSkip;
+        if (numLearnableMoves > 6) {
+            numLearnableMoves = MAX_TUTOR_MOVES_PER_PAGE;
+        }
+        showNextButton = TRUE;
+    } else {
+        // UB: showNextButton is never initialized in this branch.
+        // Fortunately, this branch is never taken, as an
+        // appropriate value of pageNum is always passed in
+        // for all instances where this function is called.
+        numLearnableMoves = MAX_TUTOR_MOVES_PER_PAGE;
+        numMovesToSkip = 0;
+    }
+    u16 *result = GetVarPointer(fieldSystem, resultVarId);
+    Window *window = FieldSysGetAttrAddr(ctx->fsys, SCRIPTENV_WINDOW);
+    *unk = ov01_021EDF78(fieldSystem, 1, 1, 0, 1, result, *messageFormat, window, ctx->msgdata);
+    messageData = NewMsgDataFromNarc(MSGDATA_LOAD_DIRECT, NARC_msgdata_msg, NARC_msg_msg_0750_bin, HEAP_ID_32);
+    String *string = String_New(0x10, HEAP_ID_32);
+    showAsTwoColumns = (numLearnableMoves + showNextButton >= 4) ? 1 : 0;
+    for (i = 0; i < numLearnableMoves; i++) {
+        ReadMsgDataIntoString(messageData, sTutorMoves[learnableMoves[i + numMovesToSkip]].move, string);
+        BufferString(*messageFormat, 0, string, 2, 1, 2);
+        BufferIntegerAsString(*messageFormat, 1, sTutorMoves[learnableMoves[i + numMovesToSkip]].cost, 2, PRINTING_MODE_RIGHT_ALIGN, TRUE);
+        MoveTutorMenu_SetListItem(*unk, showAsTwoColumns, 0xff, sTutorMoves[learnableMoves[i + numMovesToSkip]].move);
+    }
+    String_Delete(string);
+    DestroyMsgData(messageData);
+    if (showNextButton) {
+        MoveTutorMenu_SetListItem(*unk, 2, 0xff, 0xfffd);
+    }
+    MoveTutorMenu_SetListItem(*unk, 3, 0xff, 0xfffe);
+    ov01_021F6ABC(fieldSystem, 3, 7, GetVarPointer(fieldSystem, ctx->data[0]));
+    SetupNativeScript(ctx, ov01_0220305C);
+    u16 *unused = GetVarPointer(fieldSystem, ctx->data[0]); // yes, this is needed here to match
+    return TRUE;
+}
+
+static BOOL ov01_0220305C(ScriptContext *ctx) {
+    FieldSystem *fieldSystem = ctx->fsys;
+    struct UnkStruct_ov01_021EDC28 **unk = ov01_021F6B20(fieldSystem);
+    u16 *result = GetVarPointer(fieldSystem, ctx->data[0]);
+    if (*result == 0xeeee) {
+        return FALSE;
+    }
+    ov01_021EDF00(*unk);
+    return TRUE;
+}
+
+// Unused
+BOOL ScrCmd_742(ScriptContext *ctx) {
+    u16 slot = ScriptGetVar(ctx);
+    u16 move = ScriptGetVar(ctx);
+    u16 *result = ScriptGetVarPointer(ctx);
+    *result = FALSE;
+    PARTY *party = SaveArray_PlayerParty_Get(ctx->fsys->savedata);
+    Pokemon *mon = GetPartyMonByIndex(party, slot);
+    u16 *unk = AllocFromHeapAtEnd(HEAP_ID_FIELD, 0x2c);
```

Comment:
ideally I'd say make the size a bit clearer, but this entire func is unused so...